src/reducers/users.js

```diff
@@ -57,6 +63,7 @@ const requestUserRolesFailure = error => ({
 });
 
 export const actions = {
+  resetLoaded,
```

Comment:
How about a setLoaded instead that takes a new object to use in place of the existing `loaded` object?

That way we can avoid [this code](https://github.com/mixnjuice/frontend/pull/43/files/a8c6ae553547e9c9694aa1d3d49c9324c2218e37..99e5ff2b8d1e9869d58c62f119846312339a7ef5#diff-3f5cc67cba0731d032221ec5bcffd3bdR90) which is going to be hard to scale.

Reply:
I refactored to use a `users.cache`, which stores the query data by page/limit, then `users.collection` holds the currently displayed data. An API request is only made if the request isn't already in the `users.cache`. This scales without the need for the problem area you identified.